docs: add metadata to all remaining pages (#1246)
canonical/rockcraft: docs/reference/extensions/index.rst

.. meta::

>     :description: Reference documentation for all Rockcraft extensions including flask-framework, django-framework, fastapi-framework, go-framework, and spring-boot-framework.

Reviewer: ```suggestion
    :description: The reference documentation for all extensions in Rockcraft.
```
